### PR TITLE
Fix dbListFields and duplicate column names

### PR DESCRIPTION
**Pull request by onurfiliz**
Some recent changes to Presto broke our assumptions about column information. We had been assuming the first response would always include that data, which is no longer the case. This PR removes that assumption and adds test cases to verify we are handling the edge cases.

### CODE REVIEW

File: tests/testthat/test-db_query_fields.R [hunk 1]
```diff
@@ -71,7 +71,7 @@ test_that('db_query_fields works with mock', {
         request_body=paste0(
           '^SELECT \\* FROM \\(',
             '\\(SELECT 1 AS a, \'t\' AS b\\) "a"',
-          '\\) "zzz[0-9]+" LIMIT 0$'
+          '\\) "zzz[0-9]+" WHERE 1 = 0$'
```
Comment (cosinequanon) [inline tests/testthat/test-db_query_fields.R]: why is this better? it could also be `WHERE false`

Comment (onurfiliz) [reply tests/testthat/test-db_query_fields.R]: It's not necessarily better. I just mirrored the dbplyr [approach](https://github.com/tidyverse/dbplyr/blob/c0e51957a2fbeb4fe7bc93f0ed677d6d33e14828/R/dbi-s3.r#L132).
